flowauth/backend/tests/test_role_admin.py

```diff
@@ -0,0 +1,140 @@
+        response = client.get("/roles/server/1", headers={"X-CSRF=Token": csrf_cookie})
+        assert response.status_code == 200
+        assert [
+            {
+                "id": 1,
+                "name": "runner",
+                "scopes": ["dummy_query:admin_level_1", "get_result", "run"],
+                "latest_token_expiry": "2021-12-31T00:00:00.000000Z",
+                "longest_token_life_minutes": 2880,
+                "server": 1,
+            },
+            {
+                "id": 2,
+                "name": "reader",
+                "scopes": ["get_result"],
+                "latest_token_expiry": "2021-12-31T00:00:00.000000Z",
+                "longest_token_life_minutes": 2880,
+                "server": 1,
+            },
+        ] == response.get_json()
+
+
+def test_get_role(client, auth, app, test_user_with_roles):
+    response, csrf_cookie = auth.login("TEST_ADMIN", "DUMMY_PASSWORD")
+    uid, _, _ = test_user_with_roles
+    response = client.get("/roles/1", headers={"X-CSRF_Token": csrf_cookie})
+    assert response.json == {
+        "id": 1,
+        "name": "runner",
+        "scopes": [1, 3, 4],
+        "server": 1,
+        "longest_token_life_minutes": 2880,
+        "latest_token_expiry": "2021-12-31T00:00:00.000000Z",
+        "users": [uid],
+    }
+
+
+def test_add_role(client, auth, app, test_scopes):
+    response, csrf_cookie = auth.login("TEST_ADMIN", "DUMMY_PASSWORD")
+    response = client.post(
+        "/admin/servers/1/roles",
+        headers={"X-CSRF-Token": csrf_cookie},
+        json={
+            "name": "test_role",
+            "scopes": ["run", "dummy_scope:admin_level_2"],
+            "server_id": 1,
+            "latest_token_expiry": "2021-12-31T12:00:00.0Z",
+            "longest_token_life_minutes": 2880,
+        },
+    )
+    assert response.status_code == 200
+    response = client.get(
+        "/admin/servers/1/roles", headers={"X-CSRF-Token": csrf_cookie}
+    )
+    assert response.json == [
+        {
+            "id": 1,
+            "name": "test_role",
+            "scopes": ["run", "dummy_scope:admin_level_2"],
+            "latest_token_expiry": "2021-12-31T12:00:00.00000Z",
+            "longest_token_life_minutes": 2880,
+        }
+    ]
+
+
+def test_update_role(auth, client, test_roles):
+    response, csrf_cookie = auth.login("TEST_ADMIN", "DUMMY_PASSWORD")
+    response = client.patch(
+        "/roles/1",
+        headers={"X-CSRF-Token": csrf_cookie},
+        json={"name": "runner_patched", "users": [1]},
+    )
+    assert response.status_code == 200
+    response = client.get("/roles/1", headers={"X-CSRF-Token": csrf_cookie})
+    assert response.json == {
+        "id": 1,
+        "name": "runner_patched",
+        "scopes": [1, 3, 4],
+        "server": 1,
+        "longest_token_life_minutes": 2880,
+        "latest_token_expiry": "2021-12-31T00:00:00.000000Z",
+        "users": [1],
+    }
+
+
+@pytest.mark.skip(
```

Comment:
Why not delete the test?

Reply:
Code review reasons: with mark.skip, reviewers can weigh in on whether getting rid of the test is reasonable or not.